data_pipeline/gmail_data/get_gmail_data.py

```diff
@@ -46,14 +52,56 @@ def iter_link_message_thread(service: Resource, user_id: str) -> Iterable[dict]:
 
 
 def get_link_message_thread_ids(service: Resource, user_id: str) -> pd.DataFrame:
-    df_link = pd.DataFrame(iter_link_message_thread(service, user_id))
+    imported_timestamp = get_current_timestamp()
+    df_link = pd.DataFrame(iter_link_message_thread_ids(service, user_id))
+    df_link['user_id'] = user_id
+    df_link['imported_timestamp'] = imported_timestamp
     return df_link
 
 
+@backoff.on_exception(backoff.expo, TimeoutError, max_tries=10)
+def get_one_thread(service: str, user_id: str, thread_id: str) -> pd.DataFrame:
+    imported_timestamp = get_current_timestamp()
+    thread_results = service.users().threads().get(userId=user_id, id=thread_id).execute()
+
+    df_thread = pd.DataFrame()
+
+    df_thread['threadId'] = [thread_results['id']]
+    df_thread['historyId'] = [thread_results['historyId']]
+    df_thread['total_email_count'] = [len(thread_results['messages'])]
```

Comment:
minor: just adding a comment anyway, that it seems redundant since we can get the count via the messages (and it would update if add new messages). But you might come across that when looking into getting incremental data (#151)